docs/source/advanced/contrib_thunder.rst

```diff
@@ -0,0 +1,468 @@
+The added to
+`thunder/core/transforms.py <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/core/transforms.py>`_
+lines
+
+.. code-block:: python
+  :lineno-start: 1111
+
+  @torchctx
+  def _topk_prim_grad(
+      a: TensorProxy, /, k: int, dim: None | int = None, largest: bool = True, sorted: bool = True, *, out=None
+  ):
+      fwd = prims.topk(a, k, dim, largest, sorted, out=out)
+      val, idx = fwd
+
+      val_grad = get_grad(val)
+
+      a_grad = ltorch.zeros_like(a)
+      # TODO: replace with scatter once we have it.
+      # scatter_add is a prim and it relies on atomic ops.
+      a_grad = ltorch.scatter_add(a_grad, dim, idx, val_grad)
+      put_grad(a, a_grad)
+
+      return fwd
+
+
+  register_grad(pids.TOPK, _topk_prim_grad)
+
+define a grad transform for ``prims.topk``.
+This operation returns a 2-tuple in forward ``fwd = (val, idx)`` with only the first element
+being differentiable. Note that *Thunder* interleaves forward and backward computations in grad transforms.
+Take a look at the lines ``val_grad = get_grad(val)``, which extracts the in-flowing backward gradient
+for ``val``, and ``put_grad(a, a_grad)`` which sets the backward gradient for the input ``a``.
+
+Do you see that comment about the missing ``scatter``? You could be the one who implements it! :)
+
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Updating the Torch Language
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+`The Torch Language`_ operations are the "highest"-level operations and, as such, are *decomposable*.
+If the missing operation can be decomposed into already existing operations, then
+`thunder/torch/__init__.py <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/torch/__init__.py>`_
+is where its implementation is to be placed.
+
+For example, consider the `PR #100, <https://github.com/Lightning-AI/lightning-thunder/pull/100>`_ that adds
+support for the Hardswish activation function.
+The function is implemented in `thunder/torch/__init__.py <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/torch/__init__.py>`_
+
+.. code-block:: python
+    :lineno-start: 1211
+    :emphasize-lines: 1-9
+
+    @torchsymbol(torch.nn.functional.hardswish, id="torch.hardswish", is_method=False)
+    def hardswish(a: TensorProxy, /, inplace: bool = False) -> TensorLike:
+        utils.check(not inplace, lambda: f"hardswish only supports inplace=False", exception_type=NotImplementedError)
+        utils.check(
+            dtypes.is_float_dtype(a.dtype),
+            lambda: f"hardswish only supports floating point dtypes, got {a.dtype}",
+            exception_type=ValueError,
+        )
+        return a * relu6(a + 3) / 6
+
+Note the checks (*Thunder* does not support in-place operations yet) and that ``hardswish`` is a composition
+of the defined in the ``torch`` language ``relu6`` operation and the language context-specific binary operations
```

Comment:
I understand that you mean binary here in the sense that you have two arguments. But for the unaware, it would be better to change the wording here for those who think of the numeral system

Reply:
It is an advanced section after all :) Do you mean the usage of explicit algebraic op names?